### ISSUE TITLE
Documentation on `spk infra` implementation

### ISSUE THREAD

**Issue by yradsmikham** [microsoft/bedrock#735]
This is supplementary to https://github.com/yradsmikham/spk-infra/blob/master/README.md. 

The goal is go a little deeper into how `spk infra` will handle specific situations in scaffolding and generation.

**Comment by yradsmikham** [comment]
For more information, visit: https://github.com/yradsmikham/spk-infra/blob/master/spk-infra-under-the-hood.md

**Comment by andrebriggs** [comment]
@yradsmikham is there a PR adding documentation to the SPK codebase?

**Comment by yradsmikham** [comment]
@andrebriggs the only piece of documentation we have in the `spk` codebase is found here: https://github.com/CatalystCode/spk/blob/master/docs/cloud-infra-management.md. We have made sure to update the documentation as we continue to build out functionalities for `spk infra`.

Should we be migrating the "end to end" design documentation to `spk`? I imagine when we are at a more stable state with the infra design, we can integrate pieces from the North Star and GitOps design over. Thoughts?

For now, I am happy to close this issue since we are able to address (or at least acknowledge) the concerns highlighted in: https://github.com/yradsmikham/spk-infra/blob/master/spk-infra-under-the-hood.md#considerations.

**Comment by andrebriggs** [comment]
Typically there is an artifact that gets produced from a task that is on the sprint board. It would be great if our SPK codebase explained the implementation. My question would be who is the audience for https://github.com/yradsmikham/spk-infra/blob/master/spk-infra-under-the-hood.md ? Eventually these things need to be in the SPK codebase.

**Comment by yradsmikham** [comment]
@andrebriggs Understood. Do you recommend moving some of the documentation over to the spk codebase? I had initially written the documentation for internal use, but now that it's been shared publicly, I'm happy to integrate it as part of spk.

**Comment by NathanielRose** [comment]
Hey so I saw that this was closed before the I had time to chime in. My only thoughts are our use of language between the Terraform templates that we use in `source` and our language around the spk scaffolded definitions and the generated deployment templates. We need to cement our terms between each transformation state so its clear when explaining. Maybe even a diagram between transformation stages. Thoughts?
